apps/edxapp/templates/services/cms/_dc_base.yml.j2

```diff
@@ -83,6 +84,16 @@ spec:
           value: {{ service_variant }}
         - name: DJANGO_SETTINGS_MODULE
           value: {{ service_variant }}.envs.fun.docker_run
+        - name: MONGODB_HOST
+          value:  "{{ mongodb_host(env_type, trashable_env_types, deployment_stamp, endpoint_mongodb_ips, edxapp_mongodb_port, edxapp_mongodb_host) }}"
+{% if edxapp_mongodb_replicaset is defined and edxapp_mongodb_replicaset | length > 1 %}
+        - name: MONGODB_REPLICASET
+          value: {{ edxapp_mongodb_replicaset }}
+{% endif %}
+{% if edxapp_mongodb_read_preference is defined and edxapp_mongodb_read_preference | length > 1 %}
```

Comment:
put this inside the above if as we don't want it when there is no replicaset?


Actually I don't understand why you need to set these variables here since you already set them in: apps/edxapp/templates/services/cms/configs/settings.yml.j2

Reply:
oh yes I have to remove them fron the settings

Reply:
but they are better in the settings since all the other settings are there... ?

Reply:
ok

Reply:
done